### PR TITLE
Set url trailing slash

### PR DESCRIPTION
**Pull request by anurag6713**
#### Summary
`url-parse` latest version is returning `pathname: '/'` which is failing OAUTH/SAML logins.
This PR removes trailing slash while setting the url.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-34855

#### Device Information
This PR was tested on: iPhone 12 Simulator, iOS 14.4

### CODE REVIEW

File: app/mm-redux/client/client4.ts [hunk 1]
```diff
@@ -68,7 +68,7 @@ export default class Client4 {
     }
 
     setUrl(url: string) {
-        this.url = url;
+        this.url = url.replace(/\/+$/, '');
```
Comment (enahum) [inline app/mm-redux/client/client4.ts:71]: This regex allows subpaths?

Comment (anurag6713) [reply app/mm-redux/client/client4.ts:71]: Yes it just removes the trailing slash and shouldn't affect the subpaths.

`http://localhost:8065/company/mattermost/` becomes `http://localhost:8065/company/mattermost`

**Review by enahum — APPROVED** [review]

File: app/mm-redux/client/client4.ts (hunk 1 above)
Comment (enahum) [reply app/mm-redux/client/client4.ts:71]: Maybe just add it to the test ;)

**Review by lindalumitchell — APPROVED** [review]
Tested on iOS 13 and 14, and Android 11, and verified the following:
- Signin to esr.test using SAML, OAuth, LDAP
- Signin to subpath.test.mattermost.com/mattermost using SAML, OAuth, LDAP

No issues found. LGTM!

**Review by furqanmlk — APPROVED** [review]
image


image


image


image

**Pull request merged**